macros/src/lib.rs

```diff
@@ -278,13 +270,34 @@ pub fn format(ts: TokenStream) -> TokenStream {
                     let exprs = fields(&var.fields, &mut fs, &mut field_types, &mut pats);
                     let pats = quote!( { #(#pats),* } );
 
-                    let encode_discriminant = if de.variants.len() == 1 {
+                    let len = de.variants.len();
+                    let encode_discriminant = if len == 1 {
                         // For single-variant enums, there is no need to encode the discriminant.
                         quote!()
-                    } else {
+                    } else if let (Ok(_), Ok(i)) = (u8::try_from(len), u8::try_from(i)) {
                         quote!(
                             f.u8(&#i);
```

Comment:
the GitHub diff is not showing this line correctly but after PR #305 the new statements need to be `f.inner.u8(&#i)`.**k b**

Reply:
Rebased to resolve this